Tests/CSharp/src/Tests/AssetSystemTests.cs

```diff
@@ -1380,6 +1380,110 @@ public static void thirdPartyPackagedAssetIdentifierTest()
+            string testSpaceDescription = "OLY-UNITTEST-SPACEDESC-REWIND";
+            string testAssetCollectionName = GenerateUniqueString("OLY-UNITTEST-ASSETCOLLECTION-REWIND");
+            string testAssetName = GenerateUniqueString("OLY-UNITTEST-ASSET-REWIND");
+
+            var sourceAssetCollection = new Systems.AssetCollection();
+
+            // Log in
+            _ = UserSystemTests.LogIn(userSystem);
+
+            // Create 'source' space and asset collection
+            var sourceSpace = SpaceSystemTests.CreateSpace(spaceSystem, sourceSpaceName, testSpaceDescription, Systems.SpaceAttributes.Private, null, null, null);
+            {
+                CreateAssetCollection(assetSystem, sourceSpace, null, testAssetCollectionName, null, null, out sourceAssetCollection);
+                CreateAsset(assetSystem, sourceAssetCollection, testAssetName, null, null, out var sourceAsset);
+
+                var fileSource = new Systems.FileAssetDataSource
+                {
+                    FilePath = Path.GetFullPath("assets/test.json")
+                };
+                fileSource.SetMimeType("application/json");
+
+                assetSystem.UploadAssetDataOnProgress += (s, e) => LogDebug($"Uploading asset... ({e.Progress}%)");
+                using var result = assetSystem.UploadAssetData(sourceAssetCollection, sourceAsset, fileSource).Result;
+                var resCode = result.GetResultCode();
+
+                Assert.AreEqual(resCode, Systems.EResultCode.Success);
+            }
+
+            // Create 'dest' space and invoke the copy
+            var destSpace = SpaceSystemTests.CreateSpace(spaceSystem, destSpaceName, testSpaceDescription, Systems.SpaceAttributes.Private, null, null, null);
+            var destAssetCollections = new Common.Array<Systems.AssetCollection>();
+            {
+                Common.Array<Systems.AssetCollection> sourceAssetCollections = new Common.Array<Systems.AssetCollection>(1);
+                sourceAssetCollections[0] = sourceAssetCollection;
+
+                using var result = assetSystem.CopyAssetCollectionsToSpace(sourceAssetCollections, destSpace.Id, false).Result;
+                
+                Assert.AreEqual(result.GetResultCode(), Systems.EResultCode.Success);
+                destAssetCollections = result.GetAssetCollections();
+            }
+
+            // Validate the copied asset collection and its data
+	        {
+                Assert.AreEqual(destAssetCollections.Size(), 1UL);
+		        Assert.AreNotEqual(destAssetCollections[0].Id, sourceAssetCollection.Id);
+		        Assert.AreEqual(destAssetCollections[0].SpaceId, destSpace.Id);
+		        Assert.AreEqual(destAssetCollections[0].Type, sourceAssetCollection.Type);
+		        Assert.AreEqual(destAssetCollections[0].Tags.Size(), 1UL);
+
+                GetAssetsInCollection(assetSystem, destAssetCollections[0], out var destAssets);
```

Comment:
Dispose `destAssets` when finished with it.

Reply:
Addressed in latest